### PR TITLE
Build: Upgrade to Go 1.19.1

### PR DESCRIPTION
**Pull request by aknuds1**
#### What this PR does
Upgrade to build with Go 1.19.1.

#### Checklist

- [na] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pracucci** [comment]
Given golang 1.19.0 has just been released, maybe we could wait to merge this PR after 2.3.0 is cut? I would prefer to not build the next Mimir stable release with a just-released golang version.

**Comment by aknuds1** [comment]
> Given golang 1.19.0 has just been released, maybe we could wait to merge this PR after 2.3.0 is cut? I would prefer to not build the next Mimir stable release with a just-released golang version.

@pracucci I think we want to wait for a few patch upgrades to Go 1.19. I made the PR in order to prepare,golangci-lint doesn't even work yet :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -91,7 +91,7 @@
 * [ENHANCEMENT] ring: optimize shuffle-shard computation when lookback is used, and all instances have registered timestamp within the lookback window. In that case we can immediately return origial ring, because we would select all instances anyway. #2309
 * [ENHANCEMENT] Memberlist: added experimental memberlist cluster label support via `-memberlist.cluster-label` and `-memberlist.cluster-label-verification-disabled` CLI flags (and their respective YAML config options). #2354
 * [ENHANCEMENT] Object storage can now be configured for all components using the `common` YAML config option key (or `-common.storage.*` CLI flags). #2330 #2347
-* [ENHANCEMENT] Go: updated to go 1.18.4. #2400
+* [ENHANCEMENT] Go: updated to go 1.19.1. #2637
```
Comment (pstibrany) [inline CHANGELOG.md:94]: This changelog entry is for 2.3.0-rc.0 version, but PR is not changing that.

**Review by pstibrany — APPROVED** [review]
Thank you!

**Pull request merged**